wandb/sdk/launch/agent/agent.py

```diff
@@ -169,16 +171,32 @@ def __init__(self, api: Api, config: Dict[str, Any]):
             self.gorilla_supports_agents,
         )
         self._id = create_response["launchAgentId"]
-        self._name = ""  # hacky: want to display this to the user but we don't get it back from gql until polling starts. fix later
         if self._api.entity_is_team(self._entity):
             wandb.termwarn(
                 f"{LOG_PREFIX}Agent is running on team entity ({self._entity}). Members of this team will be able to run code on this device."
             )
 
+        agent_response = self._api.get_launch_agent(
+            self._id, self.gorilla_supports_agents
+        )
+        self._name = agent_response["name"]
+        if self.gorilla_supports_agents:
+            self._init_agent_run()
+
     def fail_run_queue_item(self, run_queue_item_id: str) -> None:
         if self._gorilla_supports_fail_run_queue_items:
             self._api.fail_run_queue_item(run_queue_item_id)
 
+    def _init_agent_run(self) -> None:
+        settings = wandb.Settings(silent=True, disable_git=True)
+        wandb.init(
+            project=self._project,
+            entity=self._entity,
+            settings=settings,
+            id=self._name,
+            job_type=HIDDEN_AGENT_RUN_TYPE,
```

Comment:
I'm a little confused about setting the job_type to "sweep-controller". Can we pick a more appropriate term?

Reply:
We could, but then I'd need to go update every mds function in the backend. IMO, we should keep it to one fixed name, and unfortunately for historical reasons this is it :/